sale_order_carrier_auto_assign/models/sale_order.py

```diff
@@ -1,31 +1,77 @@
 # Copyright 2020 Camptocamp SA
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
-from odoo import models
+from odoo import api, models
 
 
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    @api.onchange("partner_id", "partner_shipping_id")
+    def _add_delivery_carrier_on_partner_change(self):
+        partner = self.partner_shipping_id or self.partner_id
+        if not partner:
+            return
+        if self.company_id.carrier_on_create:
+            self._set_delivery_carrier(
+                set_delivery_line=False,
+                preserve_order_carrier=False,
+            )
+
+    def _is_auto_set_carrier_on_create(self):
```

Comment:
You can maybe change it to `property`

Reply:
Sorry, I don't understand, you mean adding a `property_x` + `company_dependent=True` style field in `sale.order`? I
 don't understand the need for that, plus in some cases (in some order states) you will always want to return False.